src/types/constants.ts

```diff
@@ -0,0 +1,61 @@
+import type { WeekDayNamesLong, WeekDayNamesShort, MonthNamesLong, MonthNamesShort } from "./main.js";
+
+const WEEK_DAYS_LONG = {
+    Monday: "Monday",
+    Tuesday: "Tuesday",
+    Wednesday: "Wednesday",
+    Thursday: "Thursday",
+    Friday: "Friday",
+    Saturday: "Saturday",
+    Sunday: "Sunday",
+} as const;
+
+const WEEK_DAYS_SHORT = {
+    Mon: "Mon",
+    Tue: "Tue",
+    Wed: "Wed",
+    Thu: "Thu",
+    Fri: "Fri",
+    Sat: "Sat",
+    Sun: "Sun",
+} as const;
+
+const MONTHS_LONG = {
+    January: "January",
+    February: "February",
+    March: "March",
+    April: "April",
+    May: "May",
+    June: "June",
+    July: "July",
+    August: "August",
+    September: "September",
+    October: "October",
+    November: "November",
+    December: "December",
+} as const;
+
+const MONTHS_SHORT = {
+    Jan: "Jan",
+    Feb: "Feb",
+    Mar: "Mar",
+    Apr: "Apr",
+    May: "May",
+    Jun: "Jun",
+    Jul: "Jul",
+    Aug: "Aug",
+    Sep: "Sep",
+    Oct: "Oct",
+    Nov: "Nov",
+    Dec: "Dec",
+} as const;
+
+const weekDaysMap = Object.fromEntries(
+    Object.entries(WEEK_DAYS_LONG).map(([longName, fullName]) => [longName.slice(0, 3) as WeekDayNamesShort, fullName])
```

Comment:
No calculations should be placed in the constants folder. Please write these things by hand.